### PR TITLE
Add support for completions within the endpoint context

### LINKED ISSUE #4944: Invalid suggestions inside the Endpoint Definition
**Description:**
When tried the completions inside an endpoint definition, suggestions works but provides invalid content

### PR DESCRIPTION
**Pull request by nadeeshaan**
scope. This includes the constraint scope and the connector init scope.

ezgif com-video-to-gif

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/parsercontext/ParserRuleCallableUnitBodyContextResolver.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    @Override
+    public ArrayList<CompletionItem> resolveItems(TextDocumentServiceContext completionContext) {
+        int tokenIndex = completionContext.get(DocumentServiceKeys.TOKEN_INDEX_KEY);
+        int endpointTokenIndex = -1;
+        String tokenString;
+        TokenStream tokenStream = completionContext.get(DocumentServiceKeys.TOKEN_STREAM_KEY);
+        Position position = completionContext.get(DocumentServiceKeys.POSITION_KEY).getPosition();
+        ArrayList<CompletionItem> completionItems = new ArrayList<>();
+        
+        while (true) {
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/parsercontext/ParserRuleCallableUnitBodyContextResolver.java]: is there a specific reason to use while? can we use for loop instead? reason we can reduce few lines.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/parsercontext/ParserRuleCallableUnitBodyContextResolver.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                endpointTokenIndex = tokenIndex;
+            }
+            tokenIndex++;
+        }
+        
+        if (endpointTokenIndex > 0) {
+            Token tokenBeforeCursor = getPreviousDefaultToken(tokenStream, tokenIndex - 1);
+            switch (tokenBeforeCursor.getText()) {
+                case "create":
+                case "<":
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/parsercontext/ParserRuleCallableUnitBodyContextResolver.java]: do we need a switch here? will there be more cases if not we can simply replace this with a if statement.

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/parsercontext/ParserRuleCallableUnitBodyContextResolver.java]: Yes, ATM we have handled only one a couple of cases. There are other edge cases we need to treat especially like this.

**Comment by hevayo** [comment]
Shall we remove functions as completion items in endpoint constraint.

**Pull request merged**